apps/prairielearn/src/pages/assessmentsSwitcher/assessmentsSwitcher.ts

```diff
@@ -0,0 +1,46 @@
+import { Router } from 'express';
+import asyncHandler from 'express-async-handler';
+
+import { run } from '@prairielearn/run';
+
+import type { NavSubPage } from '../../components/Navbar.types.js';
+import { selectAssessmentsForCourseInstanceGrouped } from '../../lib/assessment.js';
+
+import { AssessmentSwitcher } from './assessmentsSwitcher.html.js';
+
+const router = Router({
+  mergeParams: true, // Ensures that assessmentsSwitcher can retrieve req.locals.course_instance and req.params.assessment_id from the parent router
+});
```

Comment:
Nit: we already do this on most pages, so I don't think this is notable enough to call out here.

```suggestion
const router = Router({ mergeParams: true });
```